wicket/src/ui/widgets/status_view.rs

```diff
@@ -0,0 +1,176 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
Love this abstraction!

Reply:
Thanks for prompting me to think about it!